test/unit/knex.js

```diff
@@ -98,10 +99,76 @@ describe('knex', () => {
     const knexWithParams = knex.withUserParams({ userParam: '451' });
 
     expect(knexWithParams.migrate.knex.userParams).to.deep.equal({
+      isProcessingDisabled: true,
+      postProcessResponse: undefined,
       userParam: '451',
+      wrapIdentifier: undefined,
     });
   });
 
+  it('copying does not result in duplicate listeners', () => {
+    if (isNode6()) {
+      return;
+    }
+
+    const knex = Knex({
+      client: 'sqlite',
+    });
+    const knexWithParams = knex.withUserParams();
+
+    expect(knex.client.listeners('start').length).to.equal(1);
+    expect(knex.client.listeners('query').length).to.equal(1);
+    expect(knex.client.listeners('query-error').length).to.equal(1);
+    expect(knex.client.listeners('query-response').length).to.equal(1);
+
+    expect(knexWithParams.client.listeners('start').length).to.equal(1);
+    expect(knexWithParams.client.listeners('query').length).to.equal(1);
+    expect(knexWithParams.client.listeners('query-error').length).to.equal(1);
+    expect(knexWithParams.client.listeners('query-response').length).to.equal(
+      1
+    );
+  });
+
+  it('listeners added to knex directly get copied correctly', () => {
+    const knex = Knex({
+      client: 'sqlite',
+    });
+    const onQueryResponse = function(response, obj, builder) {};
+    expect(knex.listeners('query-response').length).to.equal(0);
+    knex.on('query-response', onQueryResponse);
+
+    const knexWithParams = knex.withUserParams();
+
+    expect(knex.listeners('query-response').length).to.equal(1);
+    expect(knexWithParams.listeners('query-response').length).to.equal(1);
+  });
+
+  it('adding listener to copy does not affect base knex', () => {
+    if (isNode6()) {
```

Comment:
I'm worried about this !node6 stuff. I would like to know the root cause why this could not work with it. Is babel doing something strange... maybe that polyfill thing is causing it?